mlrun/datastore/v3io.py

```diff
@@ -139,17 +139,29 @@ def _put(self, key, data, max_chunk_size: int = ONE_GB):
             return
         append_header = deepcopy(self.headers)
         append_header["Range"] = "-1"
-        buffer_offset = 0
-        buffer = memoryview(data)
-        while buffer_offset < buffer_size:
-            chunk_size = min(buffer_size - buffer_offset, max_chunk_size)
-            http_put(
-                self.url + self._join(key),
-                buffer[buffer_offset : buffer_offset + chunk_size],
-                append_header if buffer_offset else self.headers,
-                None,
-            )
-            buffer_offset += chunk_size
+        if isinstance(data, str):
```

Comment:
possibly str is not the only problematic case. how about:
try:
   data = memoryview(data)
except TypeError:
   pass
so "data" will be either the memoryview or the original "data" object
and then use "data" in the loop

Reply:
Thought about it. Would slicing  buffer[buffer_offset : buffer_offset + chunk_size] make sense then?